src/Sylius/Behat/Context/Ui/ProductContext.php

```diff
@@ -52,4 +82,26 @@ public function iShouldNotBeAbleToAccessProduct(ProductInterface $product)
 
         expect($this->productShowPage->isOpen(['product' => $product]))->toBe(false);
     }
+
+    /**
+     * @When I delete the :product product
+     */
+    public function iDeleteProduct(ProductInterface $product)
+    {
+        $this->sharedStorage->set('product', $product);
```

Comment:
But by setting it there it's id is also saved. If I placed it after the operations the id wouldn't be there. :)